apps/music/js/app.js

```diff
@@ -258,6 +258,7 @@ function setHeaderTitle(title) {
 
 function setBackButtonHidden(hidden) {
   header.els.actionButton.style.visibility = hidden ? 'hidden' : 'visible';
+  scanProgress.hidden = !hidden;
```

Comment:
No, because there are two levels of visibility: if `.hidden` is true, it's always hidden. If `.hidden` is false, then we defer to `<music-scan-progress>`'s implementation to determine when to show; if you've called `.clear()`, the element will stay hidden (because the *child* of `<music-scan-progress>` is set to hidden).

I chose to do it this way so that the `.hidden` attribute on the main element is only set by users of the element, since that keeps the logic simple. The internal state that gets determined by the current scan progress is all handled privately (in the shadow DOM).